csharp/test/Microsoft.ML.OnnxRuntime.Tests/Microsoft.ML.OnnxRuntime.Tests.csproj

```diff
@@ -38,15 +38,15 @@
       <Visible>false</Visible>
     </None>
 
-    <None Include="$(OnnxRuntimeBuildDirectory)\models\**\*">
-        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-        <Visible>false</Visible>
-    </None>
+    <BuildEnvVars Include="OnnxRuntimeBuildDirectory=$(OnnxRuntimeBuildDirectory)" />
   </ItemGroup>
 
+  <Target Name="DefineBuildEnvironmentVariables" BeforeTargets="Build">
```

Comment:
How does it releated to the "Directory.Build.props" file ?

Reply:
This target writes the OnnxRuntimeBuildDirectory value to a properties file in the output folder, so that the unit tests can pick up on the value and locate the models folder. Ideally we'd be able to pass this value in memory, but msbuild/C# doesn't allow passing project property values to code files directly (hence this indirect approach).

The Directory.Build.props file is probably used for a different purpose. I'll check with Asad and see if we can drop it. OnnxRuntimeBuildDirectory value is picked up from environment variables in the packaging pipelines, so I'm not sure why we still need that file.

Reply:
Directory.Build.props file was added by @edgchen1.

Reply:
Directory.Build.props is generated by CMake to ensure it has the correct value for OnnxRuntimeBuildDirectory.

I think relying on an environment variable from the pipeline for that path is more indirect and should be at least documented clearly if we want to do that. I remember it caused some confusing build failures for me when I tried cleaning up that variable in the internal build.

Another case to consider is if someone passes a non-default build directory to build.py. They could either remember to set the appropriate environment variable for OnnxRuntimeBuildDirectory, or have CMake generate its value.

Reply:
In the package=test step, there's no build phase (i,e. no CMake), which is why it's passed as an environment variable. 

The current setup should be fine then -- at build-test- step CMake will set correct value in Directory.Build.props, at package-test step, it's passed via environment variable by Azure pipeline. At both the build-test step and package-test steps, the correct value from csproj property will be relayed to the unit tests.

Reply:
Oh, no.... It doesn't work


**System.IO.FileNotFoundException : Could not find file 'C:\agent\_work\10\s\csharp\test\Microsoft.ML.OnnxRuntime.EndToEndTests\bin\Debug\netcoreapp2.0\Properties.txt'**

Reply:
Fixed in #479